### PR TITLE
Fix bug where explore would not reload on network connect

### PR DESCRIPTION
**Pull request by coreyfloyd**
Fixes:
https://phabricator.wikimedia.org/T126302

Also fixing this one as well:
https://phabricator.wikimedia.org/T121924

A confluence of issues fixed by the following:

- Be sure to reload table on reconnect
- All sections need to have an empty cell - this is necessary because if a section doesn't have a cell on screen, then it isn't "visible"
- Always report 1 row per section (even if empty, see above)

Also some minor refactors

### CODE REVIEW

File: Wikipedia/Localizations/qqq.lproj/Localizable.strings [hunk 1]
```diff
@@ -371,3 +371,5 @@
 "explore-random-article-heading" = "Text for 'Random article' header";
 "explore-random-article-sub-heading" = "Subtext beneath the 'Random article' header with name of language Wikipedia";
 "explore-potd-heading" = "Text for 'Picture of the day' header";
+"home-empty-section" = "Text shown when a section of the home screen has no results";
+"home-empty-section-check-again" = "Button title to relaod a section when it has no results";
```
Comment (montehurd) [inline Wikipedia/Localizations/qqq.lproj/Localizable.strings]: Typo

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd should be good to go

**Comment by bgerstle** [comment]
:+1: lgtm. lots of changes to offline stuff. will need more regression testing in QA

**Comment by montehurd** [comment]
Offline reading seems solid still :+1:

**Comment by montehurd** [comment]
Hmm travis seems stuck on this one...

**Pull request merged**